src/montagu_deploy/certbot.py

```diff
@@ -0,0 +1,87 @@
+# https://github.com/certbot/certbot/blob/v3.0.1/acme/examples/http01_example.py
+
+import os.path
+import sys
+import tarfile
+from tempfile import TemporaryFile
+
+import docker
+from constellation import docker_util
+
+# The Docker API uses Go's FileMode values. These are different from the
+# standard values, as found in eg. stat.S_IFLNK.
+# https://pkg.go.dev/io/fs#FileMode
+DOCKER_MODE_TYPE = 0x8F280000
+DOCKER_MODE_SYMLINK = 0x8000000
+
+
+def read_file(container, path, *, follow_links=False):
```

Comment:
This is similar to https://github.com/reside-ic/constellation/blob/master/constellation/docker_util.py#L184, but supports symbol links. I'll probably upstream that functionality at some point.